app/ptokens/views.py

```diff
@@ -29,39 +31,323 @@
+    ptoken = get_object_or_404(PersonalToken, pk=tokenId)
+
+    user = request.user if request.user.is_authenticated else None
+
+    if request.method == 'POST':
+        kwargs = {}
+        metadata = {}
+        if not user:
+            return JsonResponse(
+                {'error': _('You must be authenticated via github to use this feature!')},
+                status=401)
+        event_name = request.POST.get('event_name')
+        if user.profile != ptoken.token_owner_profile:
+            return JsonResponse(
+                {'error': _('You don\'t own the requested ptoken !')},
+                status=401)
+
+        if event_name == 'mint_ptoken':
+            try:
+                new_amount_minted = float(request.POST.get('amount'))
+            except ValueError:
+                return JsonResponse(
+                    {'error': _('Invalid amount!')},
+                    status=401)
+            kwargs['total_minted'] = new_amount_minted
+            metadata['amount_minted'] = new_amount_minted
+        elif event_name == 'edit_price_ptoken':
+            value = request.POST.get('value')
+
+            try:
+                new_price = float(value)
+            except ValueError:
+                return JsonResponse(
+                    {'error': _('Invalid amount!')},
+                    status=401)
+            kwargs['value'] = new_price
+            metadata['previous_price'] = float(ptoken.value)
+        elif event_name == 'tx_update':
+            kwargs['tx_status'] = request.POST.get('tx_status')
+
+        if kwargs:
+            PersonalToken.objects.filter(pk=ptoken.id).update(**kwargs)
+
+            if metadata:
+                record_ptoken_activity(event_name, ptoken, user.profile, metadata)
+
+    return JsonResponse({'error': False, 'data': ptoken.to_standard_dict()})
+
+
+@csrf_exempt
+def ptoken_redemptions(request, tokenId):
+    """List and create token redemptions"""
+    ptoken = get_object_or_404(PersonalToken, id=tokenId)
+    network = request.POST.get('network')
+    total = request.POST.get('total', 0)
+    redemptions_state = request.GET.get('state')
+
+    if request.method == 'POST':
+        if not request.user:
+            return JsonResponse(
+                {'error': _('You must be authenticated via github to use this feature!')},
+                status=401)
+        RedemptionToken.objects.create(ptoken=ptoken, network=network, total=total, redemption_requester=request.user.profile)
+
+    redemptions = RedemptionToken.objects.filter(redemption_requester=request.user.profile)
+
+    if redemptions_state in ['request', 'accepted', 'denied', 'completed']:
+        redemptions = redemptions.filter(redemption_state=redemptions_state)
+
+
+    return JsonResponse({
+        'error': False,
+        'data': [redemption.to_standard_dict() for redemption in redemptions]
+    })
+
+
+@csrf_exempt
+def ptoken_redemption(request, redemptionId):
```

Comment:
How do I know what a given `redemptionId` is? I don't see that being set or stored in `ptoken_redemptions()`. Does this
> RedemptionToken.objects.create(ptoken=ptoken, network=network, total=total, redemption_requester=request.user.profile)
create an ID for objects?